internal/crypto/engine.go

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright 2023 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package crypto provides cryptographic functions
+package crypto
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"golang.org/x/oauth2"
+
+	serverconfig "github.com/stacklok/minder/internal/config/server"
+)
+
+//go:generate go run go.uber.org/mock/mockgen -package mock_$GOPACKAGE -destination=./mock/$GOFILE -source=./$GOFILE
+
+// Engine provides all functions to encrypt and decrypt data
+type Engine interface {
+	EncryptOAuthToken(token *oauth2.Token) (string, error)
```

Comment:
Previously this method required callers to marshal the token to JSON, and then base64 the result of the encryption method. This logic has been moved inside this method which simplifies its use.